### ISSUE TITLE
[16.0] mail_debrand: broken emails and bad replacements

### ISSUE THREAD

**Issue by guestisp** [OCA/social#1258]
When using ***mail_debrand*** module, mail are sent broken, due to an unwanted encoding of text.
In example, what you see at the bottom of this issue is what we are getting with the Digest email.
as you can see, the initial part is done properly (probably that part is not handled by mail_debrand, but the body content is urlencoded (or similiar).

Also, the tag removal doesn't work with any translation different than english, in example, this line:
https://github.com/OCA/social/blob/16.0/mail_debrand/models/mail_render_mixin.py#L107
is looking for the `Powered by` string, but this only works with an english email, anything else won't match and the replacement is not done (ie: in italian we have `Fornito da`)

There is also a bug that doesn't remove the leading link text : 
https://github.com/OCA/social/blob/16.0/mail_debrand/models/mail_render_mixin.py#L30-L52

With a code like this:
```
<div>
   Powered by <a href="https://odoo.com">Odoo</a>
</div>
```

the link is removed as expected, but the leading text `Powered by` is retained, resulting in a "broken" text on the resulting email. "Powered by" who, as the name is removed ? I think text should be removed too. (i've tried to address these in my PR for 17.0)

broken email:

```
Content-Type: text/html; charset="utf-8"
Content-Transfer-Encoding: quoted-printable
MIME-Version: 1.0


<!DOCTYPE html>
<html xmlns=3D"http://www.w3.org/1999/xhtml">
    <head>
        <meta http-equiv=3D"Content-Type" content=3D"text/html; charset=3DUTF=
-8"/>
        <meta name=3D"format-detection" content=3D"telephone=3Dno"/>
        <meta name=3D"viewport" content=3D"width=3Ddevice-width; initial-scal=
e=3D1.0; maximum-scale=3D1.0; user-scalable=3Dno;"/>
        <meta http-equiv=3D"X-UA-Compatible" content=3D"IE=3D9; IE=3D8; IE=3D=
7; IE=3DEDGE"/>

        <style type=3D"text/css">
           =20
            body {
                margin: 0;
                padding: 0;
                font-family: Arial, Helvetica, Verdana, sans-serif;
            }
            #header_background {
                background-color: #000000;
            }
            .global_layout {
                max-width: 588px;
                margin: 0 auto;
                border: 1px solid #eeeeee;
                border-top: 0;
                background-color: #ffffff;
            }
            .company_name {
                display: inline;
                vertical-align: middle;
                font-weight: bold;
                color: #8f8f8f;
            }
            .title_subtitle {
                font-weight: bold;
                color: #282f33;
                word-break: break-all;
            }
            .button {
                float: right;
                background-color: #000000;
                color: #ffffff;
                border-radius: 5px;
            }
            #button_connect {
                text-align: center;
            }
            .date {
                color: #8f8f8f;
            }
            .tip_title {
                margin-top: 0;
                font-weight: bold;
            }
            .tip_content {
                margin: 0 auto;
                color:#333333;
                text-align: justify;
                text-justify: inter-word;
            }
            .tip_button {
                background-color: #000000;
                border-radius: 5px;
                margin: 14px 16px 14px 0px;
                padding: 10px;
                text-decoration: none;
            }
            .tip_button_text {
                color: #ffffff;
            }
            .illustration_border {
                width: 100%;
                border: 1px solid grey;
            }
            .kpi_row_footer {
                padding-bottom: 20px;
            }
            .kpi_header {
                overflow: auto;
                padding-bottom: 10px;
                border-bottom: 1px solid #eeeeee;
                font-size: 15px;
                font-weight: bold;
                color: #282f33;
            }
            #button_open_report {
                padding: 5px 10px;
                font-size: 12px;
                font-weight: normal;
            }
            .kpi_cell {
                float: left;
                width: 33%;
                text-align: center;
                padding-top: 2px;
            }
            .kpi_cell_center {
                border-top: 2px solid #000000;
            }
            .kpi_cell_border {
                border-top: 2px solid #00A09D;
            }
            .kpi_value {
                color: #282f33;
                font-weight: bold;
                text-decoration: none;
            }
            .kpi_center_col {
                color: #000000;
            }
            .kpi_border_col {
                color: #00A09D;
            }
            .kpi_value_label {
                display: inline-block;
                margin-bottom: 10px;
                color: #888888;
                font-size: 12px;
                text-transform: uppercase;
            }
            .kpi_margin {
                width: 75px;
                padding: 5px 10px;
                text-decoration: none;
                border-radius: 5px;
            }
            .positive_kpi_margin {
                border: 1px solid #c4ecd7;
                background-color: #d5f1e2;
                color: #17613a;
            }
            .negative_kpi_margin {
                border: 1px solid #f4cfce;
                background-color: #f7dddc;
                color: #712b29;
            }
            .kpi_trick {
                clear: both;
            }
            .download_app {
                height: 40px;
                margin-bottom: 5px;
                display: inline-block;
            }
            .preference_div {
                clear: both;
                background-color: #eeeeee;
                text-align: center;
            }
            .preference {
                margin-bottom: 15px;
                color: #6b6d70;
            }
            .preference a {
                text-decoration: none;
            }
            .by_odoo {
                color: #8f8f8f;
            }
            .odoo_link {
                text-decoration: none;
            }
            .odoo_link_text {
                font-weight: bold;
                color: #000000;
            }
            .run_business {
                color: #2d2a26;
            }
            #footer {
                background-color: #eeeeee;
                color: #8f8f8f;
                text-align: center;
            }

            @media only screen and (max-width: 650px) {
                .d-block {
                    display: block !important;
                }
                #header_background {
                    padding-top: 0px;
                }
                #header {
                    padding: 15px 20px;
                    border: 1px solid #eeeeee;
                }
                .global_layout {
                    padding: 20px;
                }
                .company_name {
                    font-size: 15px;
                }
                .title_subtitle {
                    margin: 5px auto;
                }
                #button_connect {
                    padding: 4px 10px;
                    font-size: 12px;
                }
                .date {
                    margin-top: 10px;
                    font-size: 10px;
                }
                .tip_title {
                    font-size: 14px;
                }
                .tip_content {
                    margin: 10px auto 0 auto;
                    font-size: 12px;
                    line-height: 20px;
                }
                .illustration_border {
                    margin-top: 15px;
                }
                .kpi_value {
                    font-size: 20px;
                }
                .kpi_margin {
                    font-size: 10px;
                }
                .kpi_margin_margin {
                    margin-bottom: 5px;
                }
                .preference_div {
                    padding: 15px;
                }
                .preference {
                    font-size: 12px;
                }
                .by_odoo {
                    font-size: 10px;
                }
                .run_business {
                    margin: 20px auto;
                    font-size: 12px;
                }
                #footer {
                    font-size: 15px;
                }
                #powered {
                    margin-top: 15px;
                }
            }

            @media only screen and (min-width: 651px) {
                #header_background {
                    padding-top: 70px;
                }
                #header {
                    padding: 20px 30px 25px 30px;
                    border-left: 1px solid #000000;
                    border-right: 1px solid #000000;
                }
                .global_layout {
                    padding: 25px 30px 30px 30px;
                }
                .company_name {
                    font-size: 25px;
                }
                .title_subtitle {
                    margin: 10px auto;
                }
                .button {
                    padding: 5px 10px;
                }
                #button_connect {
                    padding: 6px 10px;
                    font-size: 15px;
                }
                .date {
                    margin-top: 15px;
                    font-size: 12px;
                }
                .tip_title {
                    font-size: 20px;
                }
                .tip_content {
                    margin: 15px auto 0 auto;
                    font-size: 15px;
                    line-height: 25px;
                }
                .illustration_border {
                    margin-top: 20px;
                }
                .kpi_value {
                    font-size: 30px;
                }
                .kpi_margin {
                    font-size: 12px;
                }
                .kpi_margin_margin {
                    margin-bottom: 10px;
                }
                .preference_div {
                    padding: 20px;
                }
                .preference {
                    font-size: 15px;
                }
                .by_odoo {
                    font-size: 12px;
                }
                .run_business {
                    margin: 15px auto;
                    font-size: 18px;
                }
                #footer {
                    font-size: 20px;
                }
                #stock_tip {
                    overflow: auto;
                    margin-top: 20px;
                }
                #stock_div_img {
                    text-align: center;
                }
                #stock_img {
                    width: 70%;
                }
            }
         </style>
    </head>
    <body>
        &lt;div&gt;
&lt;div id=3D&#34;header_background&#34;&gt;
    &lt;div class=3D&#34;global_layout&#34; id=3D&#34;header&#34;&gt;
        &lt;div style=3D&#34;overflow: auto;&#34;&gt;
            &lt;p class=3D&#34;company_name&#34;&gt;My Company (San Francisco=
)&lt;/p&gt;
            &lt;a target=3D&#34;_blank&#34; href=3D&#34;http://127.0.0.1:8069=
&#34;&gt;&lt;span class=3D&#34;button&#34; id=3D&#34;button_connect&#34;&gt;C=
ollega&lt;/span&gt;&lt;/a&gt;
        &lt;/div&gt;
        &lt;div class=3D&#34;title_subtitle&#34;&gt;
            &lt;p&gt;Riepilogo periodico Odoo personale&lt;/p&gt;
        &lt;/div&gt;
        &lt;div class=3D&#34;date&#34;&gt;novembre 21, 2023&lt;/div&gt;
    &lt;/div&gt;
&lt;/div&gt;
```

**Comment by StefanRijnhart** [comment]
Please have a look at https://github.com/OCA/social/pull/1272

**Comment by StefanRijnhart** [comment]
Fixed in 16.0 with #1272
